File: python/fib.py
#!/usr/bin/env python3
import sys
import time

def fib(n):
    if n <= 1:
        return n
    return fib(n - 1) + fib(n - 2)

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("Usage: python3 fib.py <n>")
        sys.exit(1)
    
    n = int(sys.argv[1])
    start_time = time.time()
    result = fib(n)
    end_time = time.time()
    
    time_ms = int((end_time - start_time) * 1000)
    
    print(f"Python: fib({n}) = {result}")
    print(f"Time: {time_ms}ms")

File: lua/fib.lua
#!/usr/bin/env lua

function fib(n)
    if n <= 1 then
        return n
    end
    return fib(n - 1) + fib(n - 2)
end

if not arg[1] then
    print("Usage: lua fib.lua <n>")
    os.exit(1)
end

local n = tonumber(arg[1])
local start_time = os.clock()
local result = fib(n)
local end_time = os.clock()

local time_ms = math.floor((end_time - start_time) * 1000)

print("Lua: fib(" .. n .. ") = " .. result)
print("Time: " .. time_ms .. "ms")